src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -15,15 +17,54 @@ public FileContentResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            RangeItemHeaderValue range;
+            long rangeLength;
+            if (result.LastModified.HasValue)
+            {
+                (range, rangeLength) = SetHeadersAndLog(
+                    context,
+                    result,
+                    result.FileContents.Length,
+                    result.LastModified.Value,
+                    result.EntityTag);
+            }
+            else
+            {
+                (range, rangeLength) = SetHeadersAndLog(
+                    context,
+                    result,
+                    result.FileContents.Length);
+            }
+
+            var statusCode = context.HttpContext.Response.StatusCode;
+            if (statusCode == StatusCodes.Status412PreconditionFailed ||
```

Comment:
This check is redundant. SetHeadersAndLog can return a bool from the same code that sets the status codes.